docs/swagger/spec.yml

```diff
@@ -184,6 +186,259 @@ paths:
           description: Unexpected error
           schema:
             $ref: '#/definitions/Error'
+  /layers/:
+    get:
+      summary: Get list of layers the user is authorized to view
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/orderingBase'
+        - $ref: '#/parameters/pageSize'
+        - $ref: '#/parameters/page'
+      responses:
+        200:
+          description: Paginated list of layers the user is authorized to view
+          schema:
+            $ref: '#/definitions/LayerPaginated'
+    post:
+      summary: Create a new layer
+      tags:
+        - Layers
+      parameters:
+        - name: layer
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/Layer'
+      responses:
+        202:
+          description: Layer details; at this point import processes may be in-progress
+          schema:
+            $ref: '#/definitions/Layer'
+  /layers/{uuid}/:
+    get:
+      summary: Retrieve layer details
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/uuid'
+      responses:
+        200:
+          description: Info about Layer
+          schema:
+            $ref: '#/definitions/Layer'
+        404:
+          description: UUID parameter does not refer to a layer or the user is not able to view the layer it refers to
+          schema:
+            $ref: '#/definitions/Error'
+    put:
+      summary: Update a layer
+      tags:
+        - Layers
+      parameters:
+        - name: layer
+          in: body
+          required: true
+          schema:
+            $ref: '#/definitions/Layer'
+        - $ref: '#/parameters/uuid'
+      responses:
+        202:
+          description: Layer update successful (further processing needed)
+          schema:
+            $ref: '#/definitions/LayerImage'
+        204:
+          description: Update successful (no further processing needed)
+        404:
+          description: The UUID parameter does not refer to a layer or the user does not have access to the layer it refers to
+          schema:
+            $ref: '#/definitions/Error'
+        default:
+          description: Unexpected error
+          schema:
+            $ref: '#/definitions/Error'
+    delete:
+      summary: Delete a layer
+      tags:
+        - Layers
+      parameters:
+        - $ref: '#/parameters/uuid'
+      responses:
+        204:
+          description: Deletion successful (no content)
+        404:
+          description: The UUID parameter does not refer to a layer or the user does not have access to the layer it refers to
+          schema:
+            $ref: '#/definitions/Error'
+  /layers/{uuid}/images/:
```

Comment:
There is also a `/layer-images/` endpoint; this is basically a convenience endpoint to list all images associated with a particular layer. It also makes image creation easier; since LayerImages potentially have a Many-to-Many relationship with Layers, this makes it easy to create a LayerImage associated with a particular Layer in a single step.